api/tests/Feature/RolePermissionTest.php

```diff
@@ -0,0 +1,195 @@
+<?php
+
+namespace Tests\Feature;
+
+use Illuminate\Foundation\Testing\RefreshDatabase;
+use Illuminate\Foundation\Testing\WithFaker;
+use Tests\TestCase;
+
+use App\Models\Role;
+use App\Models\Team;
+use App\Models\User;
+use Database\Seeders\RolePermissionSeeder;
+
+class RolePermissionTest extends TestCase
+{
+    use RefreshDatabase;
+
+    private $user;
+    private $ownedTeam;
+    private $unownedTeam;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->seed(RolePermissionSeeder::class);
+
+        $this->user = User::create([
+            'email' => 'role-permission@test.com',
+            'sub' => 'role-permission@test.com',
+        ]);
+
+        $this->ownedTeam = Team::create([
+            'name' => 'owned-team'
+        ]);
+
+        $this->unownedTeam = Team::create([
+            'name' => 'unowned-team'
+        ]);
+    }
+
+    /**
+     * Test the Guest User Role
+     *
+     * @return void
+     */
+    public function test_guest_role()
+    {
+        $guestRole = Role::where('name', 'guest')->sole();
+        $this->user->attachRole($guestRole);
+
+        $this->assertTrue($this->user->hasRole('guest'));
+        $this->assertTrue($this->user->isAbleTo([
+            'view-any-classification',
+            'view-any-department',
+            'view-any-skill',
+            'view-any-skillFamily',
+            'view-any-publishedPoolAdvertisement',
+            'view-any-applicantCount',
+            'create-any-searchRequest',
+            'view-any-team'
+        ]));
+
+        $this->cleanup();
+    }
+
+    /**
+     * Test the Base User Role
+     *
+     * @return void
+     */
+    public function test_base_user_role()
+    {
+        $baseUserRole = Role::where('name', 'base_user')->sole();
+        $this->user->attachRole($baseUserRole);
+
+        $this->assertTrue($this->user->hasRole('base_user'));
+        $this->assertTrue($this->user->isAbleTo([
+            'view-own-user',
+            'update-own-user',
+        ]));
+
+        $this->cleanup();
+    }
+
+    /**
+     * Test the Applicant Role
+     *
+     * @return void
+     */
+    public function test_applicant_role()
+    {
+        $applicantRole = Role::where('name', 'applicant')->sole();
+        $this->user->attachRole($applicantRole);
+
+        $this->assertTrue($this->user->hasRole('applicant'));
+        $this->assertTrue($this->user->isAbleTo([
```

Comment:
This will return true if the user has any one of these permissions. Did you mean to check test if they have _all_ the permissions? If so, you can add `true` as a second arg to isAbleTo.
https://laratrust.santigarcor.me/docs/7.x/usage/roles-and-permissions.html#checking-for-roles-permissions

Reply:
hehe, good catch! Yes, I wanted to check all 😅

Reply:
okay, i added strict checks and the tests are still passing [add strict permission check to test](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5560/commits/4d0ba14bd69c545d4a53b7770c897364ac5f0fae)